back/node_watcher/src/nodeWatcher.ts

```diff
@@ -57,147 +58,155 @@ function waitLagLimit(
   });
 }
 
-async function incrementor(
-  api: ApiPromise,
-  provider: WsProvider,
-  tasks: NomidotTask[]
-): Promise<void> {
-  const blockIdentifier = process.env.BLOCK_IDENTIFIER || 'IDENTIFIER';
-  let blockIndexId = '';
-  let blockIndex = parseInt(process.env.START_FROM || '0');
-  let currentSpecVersion = api.createType('u32', -1);
-  const lastKnownBestFinalized = (
-    await api.derive.chain.bestNumberFinalized()
-  ).toNumber();
-  let lastKnownBestBlock = (await api.derive.chain.bestNumber()).toNumber();
-
-  const existingBlockIndex = await prisma.blockIndexes({
-    where: {
-      identifier: blockIdentifier,
-    },
-  });
+export async function nodeWatcher(): Promise<unknown> {
+  return new Promise((_, reject) => {
+    const provider = new WsProvider(ARCHIVE_NODE_ENDPOINT);
 
-  if (existingBlockIndex.length === 0) {
-    const result = await prisma.createBlockIndex({
-      identifier: blockIdentifier,
-      startFrom: blockIndex,
-      index: blockIndex,
-    });
+    ApiPromise.create({ provider }).then(async api => {
```

Comment:
Try adding a catch block at the end of this then: `.catch(reject)`

Reply:
This is because if something now throws inside the `then(async api=>{})` block, nothing catches it.

`.catch(reject)` will say: catch that error, and pass it into the `reject` function

Reply:
Thanks, for some reason, I tried to use `error` rather than `catch`. Now catching and resolving an error works well.